shipment_advice/models/shipment_advice.py

```diff
@@ -44,7 +46,7 @@ def _default_warehouse_id(self):
         required=True,
         states={"draft": [("readonly", False)]},
         readonly=True,
-        check_company=True,
+        check_company=False,
```

Comment:
While I understand the fix in `_default_warehouse_id`, why should we set `check_company=False` here? We still need to select a warehouse according to the current company.

Reply:
Yes, however with check_company=True, **you cannot select another warehouse without making company_id readonly=False and changing its value first**. Then, the newly selected company_id warehouses will appear on the list

tested on runboat:
![image](https://github.com/user-attachments/assets/9281274b-7793-4726-ae7e-348b3dbe642f)

Reply:
Okay, I agree, this `check_company=True` has to be removed in such case.

Another proposal: IMO we should create shipment advices the same way we could create e.g a SO or PO: if we check how it is done there, the `company_id` field is invisible and set by default with the current user's selected company (`self.env.company`), and some fields like fiscal positions are configured with `check_company=True`.
That would mean here updating current `company_id` field to a normal M2O (not a related one) with `default=self.env.company`, and then keep the `check_company=True` on `warehouse_id`. That way we align on how it is done elsewhere in Odoo.
What do you think?